nucypher/blockchain/eth/token.py

```diff
@@ -652,26 +676,23 @@ def _do_work(self) -> None:
 
         current_block_number = self.worker.client.block_number
 
-        # alt approach
-        # external tracking
-        # pending_block = self.worker.client.w3.eth.getBlock('pending', full_transactions=True)
-        # pending_block_transactions = pending_block.transactions
-        # pending_worker_transactions = [tx for tx in pending_block_transactions
-        #                                if tx['from'] == self.worker.checksum_address]
-
         # self-tracking
-        unmined_transactions = self.__track_pending_commitments(current_block_number=current_block_number)
+        unmined_transactions = self.__track_pending_commitments()
         if unmined_transactions:
-            tx_firing_block_number, txhash = list(self.pending.items())[0]
+            tx_firing_block_number, txhash = list(sorted(self.pending.items()))[0]
             self.log.info(f'Waiting for pending commitment transaction to be mined ({txhash}).')
 
             # If the transaction is still not mined after a max confirmation time (based on current gas strategy),
             # issue a replacement transaction.
             wait_time_in_blocks = current_block_number - tx_firing_block_number
             wait_time_in_seconds = wait_time_in_blocks * AVERAGE_BLOCK_TIME_IN_SECONDS
             if wait_time_in_seconds > self.max_confirmation_time:
-                self.log.info(f"We've waited for {wait_time_in_seconds}, but max time is {self.max_confirmation_time} "
-                              f"for {self.gas_strategy} gas strategy. Issuing a replacement transaction.")
+                if txhash is UNTRACKED_PENDING_TRANSACTION:
+                    message = f"We've an untracked pending transaction. Issuing a replacement transaction."
```

Comment:
Just to confirm - we can make this determination because `commit to next period` is the only worker transaction

Reply:
Yes, we're playing with fire here, actually. But if we know that we have untracked pending transactions, we don't have a way to retrieve those TX hashes.

Reply:
I think... It is possible to get the full transaction from the pending block using `getBlock(full_transactions=True)`, filtrate by sender account, and read it's function selector / get it's hash?